src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -170,30 +103,60 @@ private function createRelationsForItem($item, $formattedRelations)
             if (! isset($relationDetails['model'])) {
                 continue;
             }
-            $model = $relationDetails['model'];
             $relation = $item->{$relationMethod}();
+            $relation_type = $relationDetails['relation_type'];
+
+            switch ($relation_type) {
+                case 'BelongsTo':
+                    $modelInstance = $relationDetails['model']::find($relationDetails['values'])->first();
+                    if ($modelInstance != null) {
+                        $relation->associate($modelInstance)->save();
+                    } else {
+                        $relation->dissociate()->save();
+                    }
+                    break;
+
+                case 'HasOne':
+                case 'MorphOne':
+                        $modelInstance = $this->createUpdateOrDeleteOneToOneRelation($relation, $relationMethod, $relationDetails);
+                    break;
+                case 'HasMany':
+                case 'MorphMany':
+                    $relation_values = $relationDetails['values'][$relationMethod];
+                    // if relation values are null we can only attach, also we check if we sent
+                    // - a single dimensional array: [1,2,3]
+                    // - an array of arrays: [[1][2][3]]
+                    // if is as single dimensional array we can only attach.
+                    if ($relation_values === null || count($relation_values) == count($relation_values, COUNT_RECURSIVE)) {
+                        $this->attachManyRelation($item, $relation, $relationDetails, $relation_values);
+                    } else {
+                        $this->createManyEntries($item, $relation, $relationMethod, $relationDetails);
+                    }
+                    break;
+                case 'BelongsToMany':
+                case 'MorphToMany':
```

Comment:
I see `BelongsToMany` and `MorphToMany` are new here - they weren't treated before. What's going on here?

1) What happened before, what's different now?
2) How can I test this?

Reply:
I explained why is it here now, but resuming this was dealt in `syncPivot` that's why it wasn't here.